Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Gun.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+
+}
+
+bool Gun::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Gun::Shoot(VDir target,id_ptr_on<Human> shooter)
+{	
+    if(Handle_Ammo())
+    {
+      if(GetOwner()->GetOwner().valid())
+      {
+          id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner()->GetOwner());
+          p->MakeMovementPattern(target,shooter);
+      }
+             //       Create<Item>(Bullet_Casing::T_ITEM_S(),GetOwner);	
+    }
+    else
+    {
+          PlaySoundIfVisible("empty.ogg",owner.ret_id());
+		// the *click* text from ss13
```

Comment:
It is actually a sort of bug in the code. #264 refer to it. `id_ptr_on` can be implicit casted to `bool` and then `bool` is casted to `size_t`. So it is basically was `PlaySoundIfVisible("sound", 1)`. It will be fixed later (so it will be impossible to do such mistake), but for now it is needed to be carefully about that.